src/daemon.py

```diff
@@ -1,22 +1,32 @@
 #!/usr/bin/env python
 import logging
-import time
 import yara  # type: ignore
-from functools import lru_cache
-from yara import SyntaxError
 import config
+import json
+import sys
 from lib.ursadb import UrsaDb
-from lib.yaraparse import parse_yara, combine_rules
 from util import setup_logging
-from typing import Any, Dict, List
-from db import Database, JobId, MatchInfo
+from typing import Any, List, Dict
+from lib.yaraparse import parse_yara, combine_rules
+from db import AgentTask, JobId, Database, MatchInfo
+
 
-db = Database()
-ursa = UrsaDb(config.BACKEND)
+YARA_CACHE: Dict[str, Any] = {}
```

Comment:
I think it should be flushed somehow to prevent memory leak (seems to be a bit tricky).